airflow/providers/google/cloud/operators/vertex_ai/pipeline_job.py

```diff
@@ -173,20 +181,46 @@ def execute(self, context: Context):
             create_request_timeout=self.create_request_timeout,
             experiment=self.experiment,
         )
-
-        pipeline_job = result.to_dict()
-        pipeline_job_id = self.hook.extract_pipeline_job_id(pipeline_job)
+        pipeline_job_id = pipeline_job_obj.job_id
         self.log.info("Pipeline job was created. Job id: %s", pipeline_job_id)
-
         self.xcom_push(context, key="pipeline_job_id", value=pipeline_job_id)
         VertexAIPipelineJobLink.persist(context=context, task_instance=self, pipeline_id=pipeline_job_id)
+
+        if self.deferrable:
+            pipeline_job_obj.wait_for_resource_creation()
+            self.defer(
+                trigger=RunPipelineJobTrigger(
+                    conn_id=self.gcp_conn_id,
+                    project_id=self.project_id,
+                    location=pipeline_job_obj.location,
+                    job_id=pipeline_job_id,
+                    poll_interval=self.poll_interval,
+                    impersonation_chain=self.impersonation_chain,
+                ),
+                method_name="execute_complete",
+            )
+
+        pipeline_job_obj.wait()
+        pipeline_job = pipeline_job_obj.to_dict()
         return pipeline_job
 
+    def execute_complete(self, context: Context, event: dict[str, Any]) -> None:
```

Comment:
It seems the type is wrongly annotated

Reply:
@Lee-W Fixed, thanks for pointing that out.